apps/studio/components/ui/CodeEditor/CodeEditor.tsx

```diff
@@ -61,6 +95,21 @@ const CodeEditor = ({
     if (autofocus) editor?.focus()
   }
 
+  const onChangeContent: OnChange = (value) => {
+    hasValue.current.set((value ?? '').length > 0)
+
+    const placeholderEl = document.getElementById(placeholderId) as HTMLElement | null
```

Comment:
Can we use a state variable here instead of the imperative `getElementById`? If Monaco is causing issues with rerenders, you can always use a `useRef` to the div element.

Reply:
oh yup good point gotcha on this